### PR TITLE
Filter on all available second factor types

### PR DESCRIPTION
**Pull request by MKodde**
Previously only hard coded second factor types where shown in the filter in the overview. This change enables us to configure the value that is displayed in this select list in the `samlstepupproviders_parameters.yml`.

The Stepup-Deploy gssp installation instructions have been updated with the new configuration value.

See Pivotal https://www.pivotaltracker.com/story/show/153357335
See Stepup-Deploy [PR#50](https://github.com/OpenConext/Stepup-Deploy/pull/50)

### CODE REVIEW

File: src/Surfnet/StepupRa/RaBundle/Form/Extension/SecondFactorTypeChoiceList.php [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    {
+        $selectOptions = [];
+        $collection = $this->secondFactorTypeService->getAvailableSecondFactorTypes();
+
+        sort($collection);
+
+        foreach ($collection as $sfTypeIdentifier) {
+
+            $translationId = sprintf($this->translationIdFormat, $sfTypeIdentifier);
+            $translation = $this->translator->trans($translationId);
```
Comment (jorissteyn) [inline src/Surfnet/StepupRa/RaBundle/Form/Extension/SecondFactorTypeChoiceList.php]: Maybe this is the right moment to disable the "don't write else statements" sniff in PHPMD!

**Comment by jorissteyn** [comment]
Two more comments:

 * when there are no results, the rror message is "Cannot filter by given token type.", I expect just the message "There are currently no tokens registered that match your query."

 * the overview already displayed the name of the GSSP token, is it really required to configure the name just for the dropdown?

File: src/Surfnet/StepupRa/RaBundle/Form/Extension/SecondFactorTypeChoiceList.php (hunk 1 above)
Comment (MKodde) [reply src/Surfnet/StepupRa/RaBundle/Form/Extension/SecondFactorTypeChoiceList.php]: This was on my mind while creating this solution. It turned out `ElseExpression` was already disabled in the exclude section of the phpmd configuration.

Added the else statement to improve the understandability of the code.

**Comment by MKodde** [comment]
Thanks for the review, some commits have been added to the PR. And feedback on your comments:
* The cannot filter by given token type validation message was produced by a validation that was still hardcoded on a set of valid second factor types. This validation has been removed in 6e7d863
* The overview shows Second factor type titles where translated against the Symfony translator. This has been resolved by using the same translation logic that was used to translate the ChoiceList options.

**Comment by MKodde** [comment]
Once more, this PR is ready for review. See [pivotal](https://www.pivotaltracker.com/story/show/155975614) for the bigger picture.

**Pull request merged**